pkg/relayer/session/proof.go

```diff
@@ -258,6 +252,81 @@ func (rs *relayerSessionsManager) goProveClaims(
 		successProofs = append(successProofs, sessionTree)
 	}
 
-	failSubmitProofsSessionsCh <- failedProofs
-	proofsGeneratedCh <- successProofs
+	return successProofs, failedProofs
+}
+
+// isProofRequired determines whether a proof is required for the given session's
+// claim based on the current proof module governance parameters.
+// TODO_TECHDEBT: Once the on/off-chain loggers are unified, move this logic to
```

Comment:
Is it overkill to just pass a logger, unify it now, and add a TECHDEBT explaining why we pas sin a logger?

Reply:
The on-chain logger `cosmossdk.io/log` and off-chain one `github.com/pokt-network/poktroll/pkg/polylog` seem to be incompatible.

I'm maybe missing something but we can't pass in a logger in both on/off-chain.

Reply:
Did you confirm that this doesn't work?

I believe this is the problem `polylog` was meant to solve. cc @bryanchriswhite 

![Screenshot 2024-09-06 at 3 38 18 PM](https://github.com/user-attachments/assets/4a0ed3b9-6971-4170-bb88-0e870272107c)

Reply:
I tried, but had the following:
![image](https://github.com/user-attachments/assets/6908de42-446d-48a3-9578-dd2ed3c72fe5)

Will sync-up with @bryanchriswhite on this.

Reply:
Updated the comments to include investigation.

Reply:
https://github.com/pokt-network/polylog/issues/3